apps/machine/zynqmp_r5/platform_info.c

```diff
@@ -182,6 +202,60 @@ int platform_init(int argc, char *argv[], void **platform)
 	}
 	*platform = rproc;
 	return 0;
+
+}
+static void report_setup(unsigned int role)
+{
+	/* R5 is remote and Linux is host */
+	unsigned int i, remote, host, len;
+	void *rsc_table = get_resource_table(0, &len);
+
+	xil_printf("### OPENAMP APP CONFIGURATION REPORT ###\n");
+	xil_printf("### APP ROLE: %s ###################\n",
+		   role == RPMSG_REMOTE ? "REMOTE" : "HOST");
+	for (i = 0; i < IPI_VALS; i++) {
+		if (ipi_info[i][1] == IPI_CHN_BITMASK) {
+			host = i;
+			break;
+		}
+	}
+
+	if (i == IPI_VALS) {
+		xil_printf("### UNEXPECTED REMOTE IPI BITMASK VALUE: 0x%x\n", IPI_CHN_BITMASK);
+		return;
+	}
+
+	for (i = 0; i < IPI_VALS; i++) {
+		if (ipi_info[i][0] == POLL_BASE_ADDR) {
+			remote = i;
+			break;
+		}
+	}
+
+	if (i == IPI_VALS) {
+		xil_printf("### UNEXPECTED POLL_BASE_ADDR for REMOTE to HOST: 0x%x\n",
+			   IPI_CHN_BITMASK);
+		return;
+	}
+
+	xil_printf("### REMOTE IPI_IRQ_VECT_ID 0x%x\n", IPI_IRQ_VECT_ID);
+	xil_printf("### REMOTE POLL_BASE_ADDR 0x%x\n", POLL_BASE_ADDR);
+	xil_printf("### REMOTE IPI_CHN_BITMASK 0x%x\n", IPI_CHN_BITMASK);
+	xil_printf("### REMOTE SHARED_MEM_PA: 0x%lx\n", SHARED_MEM_PA);
+	xil_printf("### REMOTE SHARED_BUF_OFFSET: 0x%lx\n", SHARED_BUF_OFFSET);
+	xil_printf("########################################\n");
+	xil_printf("######## EXPECTED VALUES FOR HOST: #####\n");
+	xil_printf("########################################\n");
+	xil_printf("### HOST IPI_DEV_NAME: %x.ipi\n", ipi_info[host][0]);
+	xil_printf("### HOST SHM_DEV_NAME: %x.shm\n", rsc_table);
+	xil_printf("### HOST RSC_MEM_PA: 0%lx\n", rsc_table);
+	xil_printf("### HOST VRING_MEM_PA: 0%lx\n", SHARED_MEM_PA);
+	xil_printf("### HOST SHARED_BUF_PA: 0%lx\n", SHARED_MEM_PA + SHARED_BUF_OFFSET);
+	xil_printf("### HOST IPI_CHN_BITMASK: 0x%x\n", ipi_info[remote][1]);
+	xil_printf("### EXPECTED IPI VALUES FOR HOST (RPMSG USERSPACE ONLY): ######\n");
+	xil_printf("### HOST IPI_DEV_NAME: %x.ipi\n", ipi_info[host][0]);
+	xil_printf("### HOST IPI REG VALUE: <0 %d 4>\n", ipi_info[host][2] - 32);
+	xil_printf("########################################\n");
```

Comment:
Is it intentional not to have the same order of traces for the host and the remote?

Reply:
I was trying to be consistent. but there is no hard requirement from my persepctive.